File: Examples/ObjectiveC/GDPRExample/GDPRExample/SettingsViewController.m
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

#import <YandexMobileAds/YMAMobileAds.h>
#import "SettingsViewController.h"
#import "UserDefaultsKeys.h"

@interface SettingsViewController ()

@property (nonatomic, weak) IBOutlet UISwitch *userConsentSwitch;

@end

@implementation SettingsViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.userConsentSwitch.on = [[self userDefaults] boolForKey:kGDPRUserConsentKey];
}

- (IBAction)userConsentDidChange:(UISwitch *)sender
{
    [self setUserConsent:sender.isOn];
}

- (IBAction)resetUserConsent:(UIButton *)sender
{
    [self setUserConsent:NO];
    [self.userConsentSwitch setOn:NO];
    [[self userDefaults] setBool:YES forKey:kGDPRShouldShowDialogKey];
}

- (void)setUserConsent:(BOOL)userConsent
{
    [[self userDefaults] setBool:userConsent forKey:kGDPRUserConsentKey];
    [YMAMobileAds setUserConsent:userConsent];
}

- (NSUserDefaults *)userDefaults
{
    return [NSUserDefaults standardUserDefaults];
}

@end

File: Examples/YandexMobileAdsExample/YandexMobileAdsExample/GDPR/SettingsViewController.swift
/*
 * Version for iOS © 2015–2025 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

import UIKit
import YandexMobileAds

class SettingsViewController: UIViewController {
    @IBOutlet private var userConsentSwitch: UISwitch!
    @IBOutlet private var resetButton: UIButton!

    var gdprManager: GDPRUserConsentManager?

    override func viewDidLoad() {
        super.viewDidLoad()
        userConsentSwitch.isOn = gdprManager?.userConsent ?? false
        
        userConsentSwitch.accessibilityIdentifier = GDPRAccessibility.userConsentSwitch
        resetButton.accessibilityIdentifier = GDPRAccessibility.resetButton
    }

    @IBAction func userConsentDidChange(_ sender: UISwitch) {
       setUserConsent(sender.isOn)
    }

    @IBAction func resetUserConsent(_ sender: UIButton) {
        setUserConsent(false)
        userConsentSwitch.setOn(false, animated: true)
        gdprManager?.showDialog = true
    }

    private func setUserConsent(_ consent : Bool) {
        gdprManager?.userConsent = consent
    }
}
